admin/spec/controllers/pupil.spec.js

```diff
@@ -445,4 +456,68 @@ describe('pupil controller:', () => {
       done()
     })
   })
+
+  describe('#postAddPupil2 refactor', () => {
+    let sandbox, controller, nextSpy, pupilAddServiceSpy, req, res
+    let goodReqParams = {
+      method: 'POST',
+      url: '/school/pupil/add',
+      session: {
+        id: 'ArRFdOiz1xI8w0ljtvVuD6LU39pcfgqy'
+      }
+    }
+
+    beforeEach(() => {
+      sandbox = sinon.createSandbox()
+      res = getRes()
+      req = getReq(goodReqParams)
+      nextSpy = sandbox.spy()
+    })
+
+    afterEach(() => { sandbox.restore() })
+
+    describe('the pupilData is saved', () => {
+      beforeEach(() => {
+        pupilAddServiceSpy = sandbox.stub(pupilAddService, 'addPupil').resolves(pupilMock)
+        controller = proxyquire('../../controllers/pupil.js', {
+          '../services/pupil-add-service': pupilAddService
+        }).postAddPupil2
+      })
+
+      it('calls pupilAddService to add a new pupil to the database', async (done) => {
+        await controller(req, res, nextSpy)
+        expect(pupilAddServiceSpy.callCount).toBe(1)
+        done()
+      })
+
+      it('redirects to the pupil register page', async (done) => {
+        await controller(req, res, nextSpy)
+        expect(res.statusCode).toBe(302)
```

Comment:
that's just a 302 code where's the redirection call?
Perhaps also check for the page title to be what you expect?